grakn-graph/src/main/java/ai/grakn/graph/internal/TypeImpl.java

```diff
@@ -82,6 +99,15 @@
         cachedIsImplicit.set(isImplicit);
     }
 
+    @Override
+    public TypeImpl clone() {
+        try {
+            return (TypeImpl) super.clone();
+        } catch (CloneNotSupportedException e) {
+            throw new RuntimeException("Could not clone type [" + cachedTypeName.getValue() + "]", e);
+        }
+    }
```

Comment:
No. Try catch is here because it's a checked exception. This is just a temp fix anyway. Working on a deep clone mechanism right now